app/gui/src/dashboard/layouts/Drive/Categories/categoriesHooks.tsx

```diff
@@ -219,6 +222,10 @@ export function useLocalCategoryList() {
     type: 'local',
     id: 'local',
     label: getText('localCategory'),
+    /** The root path of this category. */
+    get rootPath() {
```

Comment:
I don't get why it's a getter here

Reply:
the root path of the local backend may change, and currently there's no way to reactively subscribe to it

Reply:
getter doesn't make it reactive either

Reply:
true, but we don't need it to be reactive afaik, we just need it to be correct at time of access, and this should (theoretically) do that

Reply:
Theoretically? Do we use the path somewhere,.except https://github.com/enso-org/enso/blob/21b348932a31ce6e987e611344ec7c68272ba527/app/gui/src/dashboard/layouts/Drive/directoryIdsHooks.tsx#L57 ?

Reply:
here is the main place where it is used: https://github.com/enso-org/enso/blob/21b348932a31ce6e987e611344ec7c68272ba527/app/gui/src/dashboard/pages/dashboard/Dashboard.tsx#L350

Reply:
I also don't understand why it's so complex. Why don't we store what we need in the localstorage? Why we have to do these... manipulations. We don't do anything like that when we expand folders.

Reply:
And in both places you don't need the getter

Reply:
> why it's so complex

because we need to traverse all ancestors here, not just the direct parent

afaict this is pretty difficult on the local backend because we don't have a real `parentsPath` - and it'd be a bit difficult to do this without making changes because currently directory ids on the cloud backend include slashes which conflicts with the slash delimiter used in `parentsPath`

Reply:
Can we move this logic to `LocalBackend` and `RemoteBackend`? so we'll just call a method in related backend. Also I didn't see the unification with the Path column (AFAIR it does the same but for the cloud)